libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -34,25 +34,32 @@ AP_ESC_Telem::AP_ESC_Telem()
     _singleton = this;
 }
 
-// return the average motor frequency in Hz for dynamic filtering
-float AP_ESC_Telem::get_average_motor_frequency_hz() const
+// return the average motor RPM
+float AP_ESC_Telem::get_average_rpm() const
 {
-    float motor_freq = 0.0f;
+    float rpm_avg = 0.0f;
     uint8_t valid_escs = 0;
 
-    // average the rpm of each motor and convert to Hz
+    // average the rpm of each motor
     for (uint8_t i = 0; i < ESC_TELEM_MAX_ESCS; i++) {
         float rpm;
         if (get_rpm(i, rpm)) {
-            motor_freq += rpm * (1.0f / 60.0f);
+            rpm_avg += rpm;
             valid_escs++;
         }
     }
+
     if (valid_escs > 0) {
-        motor_freq /= valid_escs;
+        rpm_avg /= valid_escs;
     }
 
-    return motor_freq;
+    return rpm_avg;
+}
+
+// return the average motor frequency in Hz for dynamic filtering
+float AP_ESC_Telem::get_average_motor_frequency_hz() const
+{
+    return get_average_rpm() / 60.0;
```

Comment:
Probably would be better if this was inlined, but I doubt it makes that much difference

Reply:
Done :+1: